proposals/crd-support.md

```diff
@@ -0,0 +1,52 @@
+* CRDs should never be deleted unless the user has explicitely asked for it. Since CRDs are global in a Kubernetes cluster, we can never know for sure that it is no longer in use.
+* CRDs should never be updated unless the user has explicitely asked for it. Updating CRDs could render them unusable for some of the services/applications that depend on them.
+* Mixing CRDs and CRs of the type in the same chart should be supported.
+
+
+## Implementation
+This proposal takes a very conservative approach on any changes to CRDs, but the hope is that it will not be to inconvenient. If CRDs are either only used by a single chart, or is used by many charts but rarely updated (like App CRD), updates should usually be possible without too much inconvenience for the user.
+
+CRDs need to be handled correctly through all the phases of a Helm release:
+
+### Install
+* CRDs should be the first resource processed during installation. CRDs must be installed into the cluster before any CRs belonging to the CRD.
+* If a chart contains one or more CRDs, the helm install process should for each of them, check if the CRD is already installed. 
+  * If it is, Helm should check if the installed CRD is identical to the one contained in the chart. 
+    * If it is, the install process can skip installing the CRD and just continue.
+    * If it is different, the install process should fail, unless the user has explicitly confirmed through a command line flag that the CRD in the chart should overwrite the existing one. If a CRD is versioned, it should allow multiple different versions of the CRD to coexist in the cluster. Overwrite will only apply if a chart has a CRD with a version that already exists.
+  * If it does not already exist, the CRD will be installed.
+* All CRDs in the chart will be tracked as part of a release when the chart is installed
+  * If the CRD was already installed, then the existing CRD will be adopted into the release.
+
+### Delete
+* When a chart is deleted, all CRDs will be orphaned, unless the user has explicitly through a command line flag decided that it should be deleted.
+ * CRs that were part of the release would be deleted with the release, even if the CRD is orphaned.
+
+### Update
+* Any new CRDs included in a new version of a chart will be handled in the same way that CRDs are handled in the install flow.
+* Any CRDs deleted in a new version of a chart will be handled in the same way that CRDs are handled in the delete flow.
+* If a CRD is already installed in the cluster, Helm should check if the CRD in the updated chart is in any way different than the existing one
+  * If it is not, then no changes are needed and the update can continue.
+  * If the CRD in the updated chart is not identical to the existing CRD, the update process needs explicit approval from the user before updating the CRD.
+
+### Rollback
+Rollback would follow the same rules as update.
```

Comment:
Here, if there is automatic addition of entries to the crd, then rollback would still need to be manual so the user knows to deal with the fallout.

Reply:
Yeah, there could be situations where an upgrade of a CRD is safe but the rollback would not be safe.